CesiumGltf/include/CesiumGltf/PropertyTexturePropertyView.h

```diff
@@ -284,173 +419,257 @@ template <typename ElementType> class PropertyTexturePropertyView {
+   * texture.
+   *
+   * If this property has a specified "no data" value, the raw value will still
+   * be returned, even if it equals the "no data" value.
+   *
+   * @param u The u-component of the texture coordinates.
+   * @param v The v-component of the texture coordinates.
+   *
+   * @return The value at the nearest pixel to the texture coordinates.
+   */
 
-    return PropertyArrayView<T>(std::move(result));
-  }
+  ElementType getRaw(double u, double v) const noexcept {
+    assert(
+        this->_status == PropertyTexturePropertyViewStatus::Valid &&
+        "Check the status() first to make sure view is valid");
 
-  double applySamplerWrapS(const double u, const int32_t wrapS) const noexcept {
-    if (wrapS == Sampler::WrapS::REPEAT) {
-      double integral = 0;
-      double fraction = std::modf(u, &integral);
-      return fraction < 0 ? 1.0 - fraction : fraction;
-    }
+    double wrappedU = applySamplerWrapS(u, this->_pSampler->wrapS);
+    double wrappedV = applySamplerWrapT(v, this->_pSampler->wrapT);
+
+    // TODO: account for sampler's filter (can be nearest or linear)
+
+    // For nearest filtering, std::floor is used instead of std::round.
+    // This is because filtering is supposed to consider the pixel centers. But
+    // memory access here acts as sampling the beginning of the pixel. Example:
+    // 0.4 * 2 = 0.8. In a 2x1 pixel image, that should be closer to the left
+    // pixel's center. But it will round to 1.0 which corresponds to the right
+    // pixel. So the right pixel has a bigger range than the left one, which is
+    // incorrect.
+    double xCoord = std::floor(wrappedU * this->_pImage->width);
+    double yCoord = std::floor(wrappedV * this->_pImage->height);
 
-    if (wrapS == Sampler::WrapS::MIRRORED_REPEAT) {
-      double integral = 0;
-      double fraction = std::abs(std::modf(u, &integral));
-      int64_t integer = static_cast<int64_t>(std::abs(integral));
-      // If the integer part is odd, the direction is reversed.
-      return integer % 2 == 1 ? 1.0 - fraction : fraction;
+    // Clamp to ensure no out-of-bounds data access
+    int64_t x = glm::clamp(
+        static_cast<int64_t>(xCoord),
+        static_cast<int64_t>(0),
+        static_cast<int64_t>(this->_pImage->width) - 1);
+    int64_t y = glm::clamp(
+        static_cast<int64_t>(yCoord),
+        static_cast<int64_t>(0),
+        static_cast<int64_t>(this->_pImage->height) - 1);
+
+    int64_t pixelIndex = this->_pImage->bytesPerChannel *
+                         this->_pImage->channels *
+                         (y * this->_pImage->width + x);
+
+    // TODO: Currently stb only outputs uint8 pixel types. If that
+    // changes this should account for additional pixel byte sizes.
+    const uint8_t* pValue = reinterpret_cast<const uint8_t*>(
+        this->_pImage->pixelData.data() + pixelIndex);
+
+    std::vector<uint8_t> channelValues(this->_channels.size());
```

Comment:
This allocation looks like it can be quite costly. The number of channels is always going to be bounded, right? Like <= 4? In that case, a `std::array<uint8_t, 4>` will be a _lot_ more efficient (no heap allocation). Can wrap it up in a span of the right size to pass to `assembleValueFromChannels` to indicate how many elements are actually used.